controllers/tensorflow/tfjob_controller.go

```diff
@@ -80,6 +82,41 @@ type TFJobReconciler struct {
 	ctrl     job_controller.JobController
 }
 
+func (r *TFJobReconciler) GetNodeForModelOutput(pods []*corev1.Pod) (nodeName string) {
+	var chiefPod *corev1.Pod
+	var masterPod *corev1.Pod
+	var worker0Pod *corev1.Pod
+
+	for _, pod := range pods {
+		rtype := pod.Labels[v1.ReplicaTypeLabel]
+		rIndex, _ := strconv.Atoi(pod.Labels[v1.ReplicaIndexLabel])
+		if rtype == strings.ToLower(string(training.TFReplicaTypeMaster)) && rIndex == 0 {
+			log.Info(fmt.Sprintf("select %s for model output", pod.Spec.NodeName))
+			masterPod = pod
```

Comment:
the expected `nodeName` can be fast returned.

Reply:
there's a preference order in which the nodeName is selected.
The logging is incorret, change that.